ecr-login/api/client.go

```diff
@@ -98,6 +99,22 @@ func (self *defaultClient) GetCredentials(registry, image string) (*Auth, error)
 	return nil, fmt.Errorf("No AuthorizationToken found for %s", registry)
 }
 
+func (self *defaultClient) ListCredentials() ([]*Auth, error) {
+	log.Debug("Listing Credentials from cache")
+
+	auths := []*Auth{}
+	for _, authEntry := range self.credentialCache.List() {
```

Comment:
I think that's the right call (returning everything). Worse case, we end up making get calls for registries that a profile hasn't accessed already, but the GetAuthorizationToken will succeed in all of these cases. Since registryId isn't a resource we conditionally evaluate, that means that you would have a token fetched for every single registry you've used in past. That's much better than the alternative of not being able to fetch a registry after switching back and forth.

I'll go ahead and go down that route and wrap this up earlier in that case (Add & Erase can wait for another commit), so we can get this fix out earlier for customers :smile: